scipy/sparse/linalg/_propack/meson.build

```diff
@@ -1,113 +1,28 @@
-  'PROPACK/double/dreorth.F',
-  'PROPACK/double/dritzvec.F',
-  'PROPACK/double/dsafescal.F',
-  'PROPACK/double/printstat.F',
+propack_sources = [
+    'PROPACK/src/common.c',
+    'PROPACK/src/gemm_overwrite.c',
+    'PROPACK/src/getu0.c',
+    'PROPACK/src/gs.c',
+    'PROPACK/src/lanbpro.c',
+    'PROPACK/src/lansvd_irl.c',
+    'PROPACK/src/lansvd.c',
+    'PROPACK/src/ritzvec.c'
 ]
 
-c_src = [
-  'PROPACK/complex8/ccdotc.F',
-  'PROPACK/complex8/cblasext.F',
-  'PROPACK/complex8/cgemm_ovwr.F',
-  'PROPACK/complex8/cgetu0.F',
-  'PROPACK/complex8/clanbpro.F',
-  'PROPACK/complex8/clansvd.F',
-  'PROPACK/complex8/clansvd_irl.F',
-  'PROPACK/complex8/creorth.F',
-  'PROPACK/complex8/critzvec.F',
-  'PROPACK/complex8/csafescal.F',
-  'PROPACK/complex8/printstat.F',
-  'PROPACK/complex8/sblasext.F',
-  'PROPACK/complex8/sbsvd.F',
-  'PROPACK/complex8/sgemm_ovwr.F'
-]
-
-z_src = [
-  'PROPACK/complex16/dblasext.F',
-  'PROPACK/complex16/dbsvd.F',
-  'PROPACK/complex16/dgemm_ovwr.F',
-  'PROPACK/complex16/printstat.F',
-  'PROPACK/complex16/zzdotc.f',
-  'PROPACK/complex16/zblasext.F',
-  'PROPACK/complex16/zgemm_ovwr.F',
-  'PROPACK/complex16/zgetu0.F',
-  'PROPACK/complex16/zlanbpro.F',
-  'PROPACK/complex16/zlansvd.F',
-  'PROPACK/complex16/zlansvd_irl.F',
-  'PROPACK/complex16/zreorth.F',
-  'PROPACK/complex16/zritzvec.F',
-  'PROPACK/complex16/zsafescal.F'
-]
-
-# Use risc msg implementation for 64-bit machines, pentium for 32-bit
-if meson.get_compiler('cpp').sizeof('void*') > 4
-  s_src += ['PROPACK/single/smgs.risc.F']
-  d_src += ['PROPACK/double/dmgs.risc.F']
-  c_src += ['PROPACK/complex8/cmgs.risc.F']
-  z_src += ['PROPACK/complex16/zmgs.risc.F']
-else
-  s_src += ['PROPACK/single/smgs.pentium.F']
-  d_src += ['PROPACK/double/dmgs.pentium.F']
-  c_src += ['PROPACK/complex8/cmgs.pentium.F']
-  z_src += ['PROPACK/complex16/zmgs.pentium.F']
-endif
-
-elements = [
-  ['_spropack', s_src, 'spropack.pyf'],
-  ['_dpropack', d_src, 'dpropack.pyf'],
-  ['_cpropack', c_src, 'cpropack.pyf'],
-  ['_zpropack', z_src, 'zpropack.pyf']
-]
-
-fargs_propack = ['-U_OPENMP']
-foreach ele: elements
-  propack_lib = static_library('lib_' + ele[0],
-    ele[1],
-    fortran_args: [
-      fargs_propack,
-      fortran_ignore_warnings,
-      _fflag_Wno_intrinsic_shadow,
-      _fflag_Wno_uninitialized,
-      _fflag_preprocess,
-    ],
-    gnu_symbol_visibility: 'hidden',
-  )
-
-  propacklib = py3.extension_module(ele[0],
-    f2py_gen.process(ele[2]),
-    link_with: propack_lib,
-    c_args: ['-U_OPENMP', _cpp_Wno_cpp],
-    fortran_args: _fflag_Wno_maybe_uninitialized,
-    dependencies: [lapack_lp64_dep, fortranobject_dep],
-    link_args: version_link_args,
-    install: true,
-    link_language: 'fortran',
-    subdir: 'scipy/sparse/linalg/_propack'
-  )
-endforeach
+propack_incdir_public = include_directories('PROPACK/include')
+propack_incdir_private = include_directories('PROPACK/src/include')
+
+propack_lib = static_library('propack_lib',
```

Comment:
Using a static library isn't needed anymore, that was only because of mixed C/Fortran sources. Here is how I'd clean up this build file further: https://github.com/rgommers/scipy/commit/42b15958ba98d3b21b3f8128e04e8159cb6c8fdc. I can push that directly if you prefer.

Reply:
Hmm that shouldn't happen normally. I just checked, in this case the sizes are almost identical, the version I proposed is 150 bytes smaller on a 390k total size (regular dev build, not release). So let's go with the simpler version:)